### ISSUE TITLE
What is the format specifier for numeric axis ticks?

### ISSUE THREAD

**Issue by kennil** [vega/vega#905]
What is the default [d3-format specifier](https://github.com/d3/d3-format/#locale_format) used, when Vega dislpays axis ticks?

My first guess was ',.0f', however this seems to ignore the definition set via [vega.formatLocale(definition)](https://vega.github.io/vega/docs/api/locale/#formatLocale) and always uses ',' as the group separator. Which is the group sepearator for en-US, d3's default.

I want to display numbers using a text mark and achieve the desired format by setting:

`"text" : {"signal": "format(datum.v, ',.0f')"}`

It's v3.0.0-beta.38. Am I missing something?

```
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "width": 100,
  "height": 200,
  "autosize": "pad",
  "data": [
    {
      "name": "table",
      "values": [
        {"v": 50000},
        {"v": 75000},
        {"v": 100000}
      ]
    }
  ],
  "scales": [
    {
      "name": "y",
      "domain": {"data": "table", "field": "v"},
      "range": "height"
    }
  ],
  "axes": [
    { "orient": "left", "scale": "y"}
  ],
  "marks": [
    {
      "type": "text",
      "from": {"data":"table"},
      "encode": {
        "enter": {
        	"x": {"signal": "width"},
        	"y": {"scale": "y", "field": "v"},
          "align": {"value": "right"},
          "baseline": {"value": "middle"},
          "text": {"signal": "format(datum.v, ',.0f')"}
        }
      }
    }
  ]
}
```

![axis_tick_format](https://user-images.githubusercontent.com/16414384/28017578-36d3a866-657a-11e7-8bb6-731bfcb1ea30.PNG)

**Comment by jheer** [comment]
Thanks for sharing the issue!

If not specified, the axis label format is automatically determined relative to the scale type. For example, for linear scales the data domain is analyzed to determine appropriate precision. The tick labels in the example you've shared are formatting correctly. To set a custom format specifier for the tick labels, use the [axis `format` property](https://vega.github.io/vega/docs/axes/). Note that if your custom specifier omits a precision setting, the precision should still be set automatically for linear, pow and sqrt scales. (If you find that the custom axis format is not working, please file an issue!)

Your issue also revealed a bug with the `format` expression function (as used in the text mark definition in your example). The expression function is using the default EN-US locale even when a custom locale has been set via the Vega API. I will commit a fix for this bug soon, which will be included in the next Vega release.

**Comment by jheer** [comment]
Fixed format expression function locale bug in vega/vega-parser@c7cea4572bb492e3037730af14338fbeb1f800e2. Will be included in Vega v3.0.0-rc2.
